pkg/infra/kubernetes/exec_unit.go

```diff
@@ -150,6 +156,33 @@ func (unit *HelmExecUnit) transformDeployment(cfg config.ExecutionUnit) ([]HelmC
 
 	deployment.Spec.Template.Spec.ServiceAccountName = unit.getServiceAccountName()
 
+	if deployment.Spec.Template.Spec.NodeSelector == nil {
+		deployment.Spec.Template.Spec.NodeSelector = make(map[string]string)
+	}
+
+	if cfg.NetworkPlacement != "" {
+		deployment.Spec.Template.Spec.NodeSelector["network_placement"] = cfg.NetworkPlacement
+	}
+	if kconfig := cfg.GetExecutionUnitParamsAsKubernetes(); kconfig.InstanceType != "" {
```

Comment:
if they dont put instance type but put DiskSizeGB how would we make the guarantee they would get enough disk space?

Reply:
we could either make every node group obey the min disk size or make one obey it and still put a node selector on